build_product_images.py

```diff
@@ -81,32 +101,41 @@ def build_image_tags(image_name, image_version, product_version):
     ]
 
 
-def build_and_publish_image(args, product):
+def build_and_publish_image(args, product) -> list[list[str]]:
     """
     Returns a list of commands that need to be run in order to build and
     publish product images.
-    """
-    commands = []
 
+    For local building, builder instances are supported.
+    """
     image_name = f'{args.registry}/stackable/{product["name"]}'
     tags = build_image_tags(image_name, args.image_version, args.product_version)
     build_args = build_image_args(product["versions"][0])
 
-    commands.append(
-        [
-            "docker",
-            "build",
-            *build_args,
-            *tags,
-            "-f",
-            product["name"] + "/Dockerfile",
-            ".",
-        ]
-    )
+    commands = [
+        "docker",
+        "buildx",
+        "build",
+        *build_args,
+        *tags,
+        "-f",
+        product["name"] + "/Dockerfile",
+        "--platform",
+        ",".join(args.architecture),
+    ]
+
     if args.push:
-        commands.append(["docker", "push", "--all-tags", image_name])
+        commands.append(
+            "--push",
+        )
+    if not args.push and len(args.architecture) == 1:
```

Comment:
Why not loud the images into local registry when build e.g. 2 images? (producing the following warning as well)
```
WARNING: No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
```

Reply:
Sadly manifest lists are currently not supported by buildx to be loaded into local registry.  You can add -d and put a --load behind a multi arch build (remove --push, both together are not working too), buildx will complain that --load can't be used for multi arch images.

Reply:
Alright thx!